core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
I think it might be okay to release at `shutdown`, actually. The executor will only be `shutdown` once the refcount reaches 0, not `shutdownNow()`, which will only reject new cancellation tasks, not interrupt those currently running. We only start a cancellation task when we start an RPC, so it should be safe to have a released executor after `shutdown = true`.